ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md

```diff
@@ -2470,6 +2471,27 @@ monitor_sessions:
       size: < bytes >
 ```
 
+#### Tap Aggregation
+
+```yaml
+tap_aggregation:
+  mode:
+    exclusive:
+      enabled: < true | false >
+      profile: < profile_name | default --> tap-aggregation-default >
+      no-errdisable:
+        - < EthernetX | Port-ChannelX >
+  encapsulation_dot1br_strip: < true | false >
+  encapsulation_vn_tag_strip: < true | false >
+  protocol_lldp_trap: < true | false >
+  # Allowed truncation_size values vary depending on the platform
+  truncation_size: < size in bytes >
+  mac_timestamp: < replace source-mac | header format 48-bit | header format 64-bit >
```

Comment:
<img width="354" alt="image" src="https://user-images.githubusercontent.com/88734745/167177392-71da0d0d-5c09-43d6-a42e-39951e8295eb.png">

Reply:
We can add a warning for mac timestamp. Although we can add three different timestamps only one will take precedence.

<img width="563" alt="image" src="https://user-images.githubusercontent.com/88734745/167177622-e22a8b72-ecf9-4c26-abd0-1c8fb4e554ab.png">

Reply:
I understand that either ```mac timestamp replace <>``` or ```mac timestamp header <>``` should be configured at a time. When we give both only **replace** is taking precedence although we can configure **header** timestamp.

Reply:
```
mac:
  timestamp:
    replace_source_mac: < true | false >
    header:
      format: < "48-bit" | "64-bit" >
      eth_type: < integer > # I found this one to be in range 0-65535
  fcs_append: < true | false >
  fcs_error: < "correct" | "discard" | "pass-through" >
```

>> Thanks! On 4.26.3M on a DCS-7280CR2A-30-F, I dont have eth-type available, so wasn't even aware of it. Do you have an info about it?

Reply:
> I understood that either `mac timestamp replace <>` or `mac timestamp header <>` should be configured. When we give both only **replace** is taking precedence although we can configure **header** timestamp.

This is correct. You can configure replace_source_mac and one header fomat (48 | 64bit) together, in this case replace source mac will take precedence.  So I think it makes sense to adjust the DM like you suggested.

To shed some more light on this from the technical point:

a) When we configure: "mac timestamp replace source-mac" the 48bit src field of the frame will be replaced with a 48bit long timestamp, the rest of the frame remains as it is

b) When we use "mac timestamp header format 64-bit" or "mac timestamp header format 64-bit", the switch will insert additional fields (that contain a timestamp, either 48 or 64 bits long) to the frame, making it bigger.

==

It wouldn't make sense to combine a) and b) at the same time, because this would mean: Replace Source MAC field with a timestamp AND (in addition) add more fields to the frame that also include another timestamp.. we would end up with a frame that doesn't have source MAC info anymore, but has 2 timestamps (redundant).

@ClausHolbechArista fell free to correct me if the explanation above is incorrect

Reply:
@tgodaA changes added. Do you have any documentation about eth_type? I cant find any. Would like to read it out of interest.
Thanks!

Reply:
Hi @ccsnw, please use the below references for mac timestamps:

1) https://www.arista.com/assets/data/pdf/Whitepapers/Overview_Arista_Timestamps.pdf
2) https://aristanetworks.force.com/AristaCommunity/s/article/analyzing-packet-header-timestamps-in-wireshark
3) https://aristanetworks.force.com/AristaCommunity/s/article/timestamping-on-the-7150-series
4) https://damianzaremba.co.uk/2017/12/decoding-arista-ethertype-headers-with-gopacket/

In ref #1 page 3 should have your required information.

Reply:
thanks!